pallets/liquidity-pools-gateway-queue/src/lib.rs

```diff
@@ -231,10 +259,12 @@ pub mod pallet {
 
 				weight_used.saturating_accrue(weight);
 
-				MessageQueue::<T>::remove(nonce);
+				MessageQueue::<T>::remove(last_processed_nonce);
 
 				// 1 write for removing the message
 				weight_used.saturating_accrue(T::DbWeight::get().writes(1));
```

Comment:
Nit: We need to add another write: `MessageQueue::remove` + `LastProcessedNonce::set`.